### ISSUE TITLE
New error is trying to reconstruct data

### ISSUE THREAD

**Issue by iandobbie** [DiamondLightSource/PySIMRecon#53]
Made it further this time, but still having trouble. I assume the error is the zoom_factors is not a list or array so you cannot index into it. 

```
(pysim) T:\OMX-data>sim-recon -c t:\OMX-data\config.ini -d macro-DM1A488si_14.dv
INFO:sim_recon.main:Loading configurations from t:\OMX-data\config.ini...
INFO:sim_recon.files.config:Configured OTFs found:

INFO:sim_recon.main:Starting reconstructions...
ERROR:sim_recon.recon:Error occurred for macro-DM1A488si_14.dv
Traceback (most recent call last):
  File "C:\Users\idobbie1\src\PySIMRecon\src\sim_recon\recon.py", line 251, in run_reconstructions
    if stitch_channels and zoom_factors.count(zoom_factors[0]) != len(
                                              ~~~~~~~~~~~~^^^
IndexError: list index out of range
SIM data files: 100%|█████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 100.06file/s]
```

**Comment by iandobbie** [comment]
In fact zoom_factors is an empty list as this is a single wavelength input file, the real root cause is this code
```
....
    stitch_channels: bool = True,
.... 

                  if stitch_channels and zoom_factors.count(zoom_factors[0]) != len(
                        zoom_factors
                    ):
                        logger.warning(
                            "Unable to stitch files due to mismatched zoom factors between wavelengths"
                        )
                        stitch_channels = False

                    if stitch_channels:
                        # Stitch channels (if requested and possible)
 ```

So the assumption is that the file is multi wavelength and if it isn't the check should set sticth_channels to False, but it fails as the zoom_factors list is empty so you cant reference zoom_factors[0]

**Comment by thomasmfish** [comment]
Hi Ian,

That's a problem that occurs when there were no valid wavelengths to process. I need to add some error handling there, because that's an incidental error that arises. Do you definitely have the OTF set correctly? That's the most likely issue. It has to be set with the same emission wavelength as the one we're processing (I know it's not in the case of the OMX test data).

Tom
________________________________

**Comment by iandobbie** [comment]
Ah, ok I didnt realise that the OTF had to match. I guess I need to go and reset the otf wavelength.

**Comment by thomasmfish** [comment]
Only in how it is configured, not in the OTF metadata. This can be specified via the --otf argument or in the main config file (probably best for keeping a record), depending on how you're setting this up. In the email where I first sent over the configs (which I think you couldn't access them from) I added some notes about how to make them work for you (mostly just customising paths).

**Comment by iandobbie** [comment]
However, maybe there should a way to apply the "wrong" otf to a file as you may want to do this.

**Comment by thomasmfish** [comment]
There is, you just need to link it properly in the config

**Comment by iandobbie** [comment]
Ok, now I see how it works. Still cant get the data to process, I dont understand it.

**Comment by thomasmfish** [comment]
I'm working on improving the exception handling, but for now would it be helpful to have a meeting? I may be able to troubleshoot

**Comment by iandobbie** [comment]
Ok, We finally got to the bottom of this. I was using cudasirecon version 1.1.1 and the options avaliable are different. Issues solved by installing version 1.2 with 

conda install -c conda-forge cudasirecon=1.2

Reconstructions are now finally working on my windows setup, Yeah!

**Comment by iandobbie** [comment]
Closing as issue is solved
